src/engine/strat_engine/setup.rs

```diff
@@ -49,48 +97,8 @@ pub fn find_all() -> EngineResult<HashMap<PoolUuid, HashMap<Device, PathBuf>>> {
             }
         };
 
-        let f = OpenOptions::new().read(true).open(&devnode);
-
-        // There are some reasons for OpenOptions::open() to return an error
-        // which are not reasons for this method to return an error.
-        // Try to distinguish. Non-error conditions are:
-        //
-        // 1. ENXIO: The device does not exist anymore. This means that the device
-        // was volatile for some reason; in that case it can not belong to
-        // Stratis so it is safe to ignore it.
-        //
-        // 2. ENOMEDIUM: The device has no medium. An example of this case is an
-        // empty optical drive.
-        //
-        // Note that it is better to be conservative and return with an
-        // error in any case where failure to read the device could result
-        // in bad data for Stratis. Additional exceptions may be added,
-        // but only with a complete justification.
-        if f.is_err() {
-            let err = f.unwrap_err();
-            match err.kind() {
-                ErrorKind::NotFound => {
-                    continue;
-                }
-                _ => {
-                    if let Some(errno) = err.raw_os_error() {
-                        match Errno::from_i32(errno) {
-                            Errno::ENXIO | Errno::ENOMEDIUM => continue,
-                            _ => return Err(EngineError::Io(err)),
-                        };
-                    } else {
-                        return Err(EngineError::Io(err));
-                    }
-                }
-            }
-        }
-
-        let mut f = f.expect("unreachable if f is err");
-        if let DevOwnership::Ours(pool_uuid, _) = StaticHeader::determine_ownership(&mut f)? {
-            // No value should ever be ejected, because duplicate device nodes
-            // are filtered out above. Therefore, the return value of insert()
-            // might as well be ignored.
-            let _ = pool_map
+        if let Some(pool_uuid) = is_stratis_device(&devnode)? {
+            pool_map
```

Comment:
For clarity, this block and the block above could become:

```
match devnode_to_devno(&devnode)? {
  None => continue,
  Some(devno) => {
    if devno_set.insert(devno) {
      is_stratis_device(&devnode).and_then(|pool_uuid| pool_map.entry(pool_uuid).or_insert_with(...).insert(...))
    } else {
      continue;
    }
  }
}
```

That gets rid of any need for the comment that was in the previous version, because it is now very clear what is happenng and the code that separated these two has now been moved into ```is_stratis_device```.